### PR TITLE
[CARBONDATA-1751] Make the type of exception and message correctly

### PR DESCRIPTION
**Pull request by xubo245**
carbon printout improper error message, for example, it printout system error when users run create table with the same column name, but it should printout related exception information
So we modify sys.error method to AnalysisException when uses run related operation except IUD,compaction and alter

Make the type of exception and message correctly. This PR focus on Spark2 module

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
    change old test cases to adapt changed massage
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
MR55, MR64

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1235/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1255/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1256/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1258/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala [hunk 1]
```diff
@@ -87,8 +88,10 @@ case class AlterTableCompactionCommand(
     } catch {
       case e: Exception =>
         if (null != e.getMessage) {
+          // TODO: We should modify it when we support compaction later.
           sys.error(s"Compaction failed. Please check logs for more info. ${ e.getMessage }")
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala]: Why not change this also?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1265/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1268/

**Comment by xubo245** [comment]
Please review it @jackylk

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1283/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1284/

**Comment by xubo245** [comment]
I have fixed conflicts, please review it @jackylk

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1302/

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala [hunk 2]
```diff
@@ -18,12 +18,10 @@
 package org.apache.spark.carbondata
 
 import scala.collection.mutable
-
```
Comment (chenerlu) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala]: Suggest keep this space line

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala [hunk 3]
```diff
@@ -18,12 +18,10 @@
...
 import org.apache.spark.sql.common.util.Spark2QueryTest
 import org.apache.spark.sql.types._
-import org.apache.spark.sql.{Row, SaveMode}
+import org.apache.spark.sql.{AnalysisException, Row, SaveMode}
 import org.scalatest.BeforeAndAfterAll
-
```
Comment (chenerlu) [inline integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala]: Suggest keep this space line

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala (hunk 2 above)
Comment (xubo245) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala]: OK, Done

File: integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala (hunk 3 above)
Comment (xubo245) [reply integration/spark2/src/test/scala/org/apache/spark/carbondata/CarbonDataSourceSuite.scala]: OK, Done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1306/

**Comment by xubo245** [comment]
Erlu Chen and Yadong Qi have reivewed this PR, please review it again @jackylk

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1771/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1772/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1773/

**Comment by jackylk** [comment]
LGTM

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1779/

**Pull request closed without merging**